index.js

```diff
@@ -1,70 +1,184 @@
 const bunyan = require('bunyan');
 
 module.exports = (options) => {
-    if (!options) options = {};
+  if (!options) options = {};
 
-    if (typeof options.app === 'undefined') options.app = {name: options.app_name};
-    if (typeof options.app.name === 'undefined') options.app.name = 'app';
-    if (typeof options.local_level === 'undefined') options.local_level = process.env.LOG_LEVEL_LOCAL;
-    if (typeof options.remote_level === 'undefined') options.remote_level = process.env.LOG_LEVEL_REMOTE;
+  if (typeof options.app === 'undefined') options.app = {name: options.app_name};
+  if (typeof options.app.name === 'undefined') options.app.name = 'app';
+  if (typeof options.local_level === 'undefined') options.local_level = process.env.LOG_LEVEL_LOCAL;
+  if (typeof options.remote_level === 'undefined') options.remote_level = process.env.LOG_LEVEL_REMOTE;
 
-    const logger = new Logger(options.app, options.local_level, options.remote_level, options.remote_auth);
+  const logger = new Logger(options.app, options.local_level, options.remote_level, options.remote_auth);
 
-    return logger.logger;
+  return logger.logger;
 };
 
 class Logger {
-    constructor(appParams, localLevel, remoteLevel, remoteAuth) {
-        this.streams = [];
+  constructor(appParams, localLevel, remoteLevel, remoteAuth) {
+    this.modules = [];
+    this.defaults = {
+      appParams,
+      localLevel,
+      remoteLevel,
+      remoteAuth,
+    };
+    this.logger = this.createLogger(appParams, localLevel, remoteLevel, remoteAuth);
+  }
 
-        let options = {};
-        if (localLevel) {
-            this.addLocalStream(localLevel);
+  createLogger(appParams, localLevel, remoteLevel, remoteAuth, name) {
+    const options = {
+      name: appParams.name,
+      streams: this.createStreams(appParams, localLevel, remoteLevel, remoteAuth),
+    };
+    if (name) {
+      options.module = name;
+    }
+    const logger = bunyan.createLogger(options);
+    logger.log = myLoggerMethod(logger, 'info');
+
+    ['trace', 'debug', 'info', 'warn', 'error', 'fatal'].forEach(logMethod => myLoggerMethod(logger, logMethod));
+    logger.module = (moduleName) => {
+      const moduleLogger = this.createLogger(appParams, localLevel, remoteLevel, remoteAuth, moduleName);
+      this.modules.push(moduleLogger);
+      return moduleLogger;
+    };
+    logger.setLevels = logger.levels; // original bunyan method
+    logger.levels = (level, module, stream) => this.levels(level, module, stream);
+    return logger;
+  }
+
+  levels(level, moduleName, streamName) {
```

Comment:
I'm getting `(node:950) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit` when calling levels() method. Also cant change log levels.

extra-dependencies:
    "@google-cloud/logging": "^4.0.1",
    "@google-cloud/logging-bunyan": "^0.9.1",
    "bunyan-buddy": "^1.5.1" <-- fixes missing 1.6.0 version error for me

Overrided modules in app/lib: 
`bunyan-buddy` (this PR's version)
`log` (uses local ../bunyan-buddy)